build.gradle

```diff
@@ -106,6 +105,7 @@ subprojects {
             version = project(":").version
         }
     }
+    apply from: "${rootDir}/gradle/publish.gradle"
```

Comment:
The reason we made publish.gradle work on gradle-plugin is so we can load gradle-plugin classpath in test-plugin, why making this change?

Reply:
Pushing it before will result in the task “module” to be null, effectively causing issues where the order is not properly respected.

Reply:
I see, then it suppose to do a check if the project has module task available

Reply:
That won’t work as the module task is injected on apply, which only happens after the module gradle thing is loaded in. The publish plugin has to initialised after the module one

Reply:
How about something like this
```gradle
    if (project.name != "gradle-plugin") {
        apply from: "${rootDir}/gradle/extensions.gradle"
        if (project.name.contains("bootstrap")) {
            apply from: "${rootDir}/gradle/bootstrap.gradle"
        } else {
            apply from: "${rootDir}/gradle/basic.gradle"
            if (project.name.endsWith("-platform")) {
                apply from: "${rootDir}/gradle/platform.gradle"
            } else if (!project.name.endsWith("-tests")) {
                apply from: "${rootDir}/gradle/module.gradle"
            } else {
                version = project(":").version
            }
        }
    }

    apply from: "${rootDir}/gradle/publish.gradle"
```

Reply:
Oh yeah that's perfect